examples/linear_model/plot_tweedie_regression_insurance_claims.py

```diff
@@ -0,0 +1,593 @@
+# on the training set via a quasi-Newton solver: l-BFGS. Some of the features
+# are collinear, we use a weak penalization to avoid numerical issues.
+glm_freq = PoissonRegressor(alpha=1e-3)
+glm_freq.fit(X_train, df_train["Frequency"],
+             sample_weight=df_train["Exposure"])
+
+scores = score_estimator(
+    glm_freq,
+    X_train,
+    X_test,
+    df_train,
+    df_test,
+    target="Frequency",
+    weights="Exposure",
+)
+print(scores)
+
+##############################################################################
+#
+# We can visually compare observed and predicted values, aggregated by the
+# drivers age (``DrivAge``), vehicle age (``VehAge``) and the insurance
+# bonus/malus (``BonusMalus``).
+
+fig, ax = plt.subplots(ncols=2, nrows=2, figsize=(16, 8))
+fig.subplots_adjust(hspace=0.3, wspace=0.2)
+
+plot_obs_pred(
+    df=df_train,
+    feature="DrivAge",
+    weight="Exposure",
+    observed="Frequency",
+    predicted=glm_freq.predict(X_train),
+    y_label="Claim Frequency",
+    title="train data",
+    ax=ax[0, 0],
+)
+
+plot_obs_pred(
+    df=df_test,
+    feature="DrivAge",
+    weight="Exposure",
+    observed="Frequency",
+    predicted=glm_freq.predict(X_test),
+    y_label="Claim Frequency",
+    title="test data",
+    ax=ax[0, 1],
+    fill_legend=True
+)
+
+plot_obs_pred(
+    df=df_test,
+    feature="VehAge",
+    weight="Exposure",
+    observed="Frequency",
+    predicted=glm_freq.predict(X_test),
+    y_label="Claim Frequency",
+    title="test data",
+    ax=ax[1, 0],
+    fill_legend=True
+)
+
+plot_obs_pred(
+    df=df_test,
+    feature="BonusMalus",
+    weight="Exposure",
+    observed="Frequency",
+    predicted=glm_freq.predict(X_test),
+    y_label="Claim Frequency",
+    title="test data",
+    ax=ax[1, 1],
+    fill_legend=True
+)
+
+
+##############################################################################
+#
+# According to the observed data, the frequency of accidents is higher for
+# drivers younger than 30 years old, and it positively correlated with the
+# `BonusMalus` variable. Our model is able to mostly correctly model this
+# behaviour.
+#
+# Severity Model -  Gamma distribution
+# ------------------------------------
+# The mean claim amount or severity (`AvgClaimAmount`) can be empirically
+# shown to follow approximately a Gamma distribution. We fit a GLM model for
+# the severity with the same features as the frequency model.
+#
+# Note:
+#
+# - We filter out ``ClaimAmount == 0`` as the Gamma distribution has support
+#   on :math:`(0, \infty)`, not :math:`[0, \infty)`.
+# - We use ``ClaimNb`` as `sample_weight`.
+
+mask_train = df_train["ClaimAmount"] > 0
+mask_test = df_test["ClaimAmount"] > 0
+
+glm_sev = GammaRegressor(alpha=10., max_iter=10000)
+
+glm_sev.fit(
+    X_train[mask_train.values],
+    df_train.loc[mask_train, "AvgClaimAmount"],
+    sample_weight=df_train.loc[mask_train, "ClaimNb"],
```

Comment:
```suggestion
    sample_weight=df_train.loc[mask_train, weights],
```

Reply:
Similar remark here. Different models have different weights. So better keep the column name explicit to make it easier to understand the modeling by just reading the code of the individual `fit` calls.